File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> array) {
    int answer = 0;
    int maxC = 0;
    vector<pair<int, int>> v(1001, {0, 0});
    
    for (int i = 0; i < array.size(); i++) {
        v[array[i]].first = array[i];
        v[array[i]].second++;
    }
    
    for (int j = 0; j < v.size(); j++) {
        if (v[j].second > maxC) {
            maxC = v[j].second;
            answer = v[j].first;
        }
        else if (v[j].second == maxC && v[j].second != 0)
            answer = -1;
    }
    return answer;
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cs
using System;

public class Solution {
    public int solution(int[] array) {
        int answer = 0;
        int max = 0;
        bool isDuplicated = false;
        int[] count = new int[1001];

        foreach (int n in array)
        {
            count[n]++;
        }

        for (int i = 0; i < count.Length; i++)
        {
            if (count[i] > max)
            {
                max = count[i];
                answer = i;
                isDuplicated = false;
            }
            else if (count[i] == max)
            {
                isDuplicated = true;
            }
        }

        if (isDuplicated)
            return -1;

        return answer;
    }
}